### DISCUSSION TITLE
Can I use as a same type Data Transformation

### DISCUSSION THREAD

**Discussion by BicycleMark** [riok/mapperly#1473] [General]
Hello Mapperly users:

We use the simple features of Mapperly to type cast from one type to another type. it works flawlessly. Can I invoke a Mapping when the two fields are the same name and same type and do a Data Transformation at that time?

Kind Regards,
Mark Wardell

**Comment by latonz** [comment]
You can either use [`MapProperty.Use`](https://mapperly.riok.app/docs/configuration/mapper/#user-implemented-property-mappings) for this (if only for selected members). If you want the custom method to be called for all member mappings for the given types you can implement a [user-implemented mapping](https://mapperly.riok.app/docs/configuration/user-implemented-methods/). This also works if the target type is the same as the source type.

**Reply by BicycleMark** [reply]
Awsome!!!

**Reply by BicycleMark** [reply]
Works great!
